src/pages/SubmissionConfig.js

```diff
@@ -25,19 +26,21 @@ export const SubmissionConfig = ({ methodId }) => {
 
   // Options chosen on this page:
   const [selectedRecordType, setSelectedRecordType] = useState()
-  const [selectedRecords, setSelectedRecords] = useState()
+  const [selectedRecords, setSelectedRecords] = useState({})
   const [configuredInputDefinition, setConfiguredInputDefinition] = useState()
   const [configuredOutputDefinition, setConfiguredOutputDefinition] = useState()
 
+  const [cachedInputSources, setCachedInputSources] = useState()
```

Comment:
this allows the user to navigate away from the Inputs tab without losing their selections

Reply:
this might have just been a dumb moment on my part - I'll either just read from the configuredInputDefinition, or I'll come back and explain why this second state is needed

Reply:
I was pleasantly surprised to find that this was *not* just me being dumb, but it may have been unnecessary:

These states cache the input sources, not just for each row, but *within* that row. In other words: if they specify a `literal` value for an attribute, and then they change that attribute to `record_lookup`, they can change back to `literal` and their original value will be preserved).